### ISSUE TITLE
How do I generate the "_dual_frip_valid_cells.txt" file that appears in the "02.Clustering.ipynb" file ?

### ISSUE THREAD

**Issue by linkous123** [Xieeeee/Droplet-Paired-Tag#2]
Hi,

Thank you for your wonderful work.

I am attempting to reproduce your code, but I can't find the "_dual_frip_valid_cells.txt" file referenced in the "02.Clustering.ipynb" file. Are the "_dual_frip_valid_cells.txt" files generated by filtering cell barcodes for both RNA and DNA? Could you please direct me to where I can find the script?

Thank you.

the code
### filter raw matrix with PF nuclei from both modalities
mtx_list <- list()
for (f in rna_source$sample){
    counts <- Read10X(paste0(path, f, "/outs/raw_feature_bc_matrix"))
    ff <- list.files(path = paste0(path, "05.R/PF_cells/"), pattern = paste0(f, ".*_dual_frip_valid_cells.txt")) ### pass-filtered barcodes
    dual_pf <- read.table(paste0(path, "05.R/PF_cells/", ff), header = T, sep = "\t")
    counts <- counts[, dual_pf$rna_bc]
    colnames(counts) <- paste0(f, ":", colnames(counts))
    mtx_list[[f]] <- counts
}

**Comment by Xieeeee** [comment]
Hi linkous123,
Thanks for your interest in Droplet Paired-Tag. The `dual_frip_valid_cells.txt` is a four column file in format like this:
```
dna_bc  rna_bc  dna_counts      rna_counts
AAACAAGCAACGTGGT-1      GGCTAGTGTAATCACG-1      813     688
AAACCAGGTAAGGTGC-1      GTTTAACCAATTGACT-1      1541    10048
```
Since we ran cellranger-atac and cellranger-rna and also do the filtering separately, we need to pair them to filter out low quality cells (cells with low reads from both modality). You can pair the barcodes from DNA and RNA using `PairArc` , then plot and select the cutoff using `PlotArc` from the `DPT_help.R`. For the pairing relationship, you can find it as [arc_bc-translation.txt](https://github.com/Xieeeee/Droplet-Paired-Tag/blob/main/01.pre-process/supp/arc_bc-translation.txt)

**Comment by AndrewHardigan** [comment]
Hello ! thanks for a great resource paper and sharing your code. I had a similar related question to poster above.

I am trying to figure out exactly what steps you took to pair and filter the paired barcodes.  I understand the general framework (run cell ranger count to generate raw bc matrix,  and cell ranger atac to generate possorted_bam.bam which is then cleaned up, peaks called and clean fragment and FRIP fragment files generated).

I have been able to follow along using the DPT.help.R functions to run ImportArcFRiP, PlotArcFRiP, and PlotArcFragment. This generated a Sample_PFcells.txt file .

My question is in the PairArc function. What are you using as input ? i.e. what is rmat and dmat?

rmat I assume is just the cellranger count raw_feature_bc_matrix.h5. However, are you using raw_peak_bc_matrix.h5 from cellranger atac as the dmat? this would allow you to run the colsums part of the PairArc function as written. However, these peaks are not the ones we have called with macs2 in "08.proc_10Xarc_DNA.sh" and would instead be called by cellranger atac.  

Or are you just taking the total counts for each cell barcode from PF_cells.txt file and using that somehow? 

Finally, are you doing two separate cutoff steps (i.e. FRIP and fragment # cutoff as seen in the first panel for each mark of Sup Fig3A, generated from PlotArcFRIP) AND a total RNA read / total fragment count cutoff (generated from PlotArcPair)? 

Apologize for my confusion.  just trying to find file inputs and expected output to generate the dual_frip_valid_cells.txt .

Thank you again for your help and this resource
